API/Controllers/ClientController.cs

```diff
@@ -0,0 +1,71 @@
+﻿using API.ViewModels;
+using API.ViewModels.CreateViewModels;
+using BLL.Services.Interfaces;
+using DAL.Models;
+using Mapster;
+using Microsoft.AspNetCore.Mvc;
+using System.Net;
+
+namespace API.Controllers;
+
+[ApiController]
+[Route("api/client")]
+public class ClientController(IClientService service) : ControllerBase
+{
+    [HttpGet]
+    [ActionName("GetAllClientsInRange")]
+    public async Task<IEnumerable<ClientViewModel>> GetAll([FromQuery] int page, [FromQuery] int pageSize, CancellationToken cancellationToken)
+    {
+        var clients = await service.GetRangeAsync(page, pageSize, cancellationToken);
+
+        var clientsVMs = clients.Adapt<IEnumerable<ClientViewModel>>();
+
+        return clientsVMs;
+    }
+
+    [HttpGet("{id}")]
+    [ActionName("GetClientById")]
+    public async Task<ClientViewModel> GetById([FromRoute] Guid id, CancellationToken cancellationToken)
+    {
+        var client = await service.GetByIdAsync(id, cancellationToken);
+
+        var clientVM = client.Adapt<ClientViewModel>();
+
+        return clientVM;
+    }
+
+    [HttpPost]
+    [ActionName("CreateClient")]
+    public async Task<ClientViewModel> Create([FromBody] CreateClientViewModel createClientViewModel, CancellationToken cancellationToken)
+    {
+        var createClientModel = createClientViewModel.Adapt<ClientModel>();
+
+        var newClient = await service.AddAsync(createClientModel, cancellationToken);
+
+        var clientVM = newClient.Adapt<ClientViewModel>();
+
+        return clientVM;
+    }
+
+    [HttpPut("{id}")]
+    [ActionName("UpdateClientById")]
+    public async Task<ClientViewModel> Update([FromRoute] Guid id, [FromBody] CreateClientViewModel updateClientViewModel, CancellationToken cancellationToken)
```

Comment:
Better to rename CreateClientViewModel to ShortClientViewModel  or UpdateClientViewModel